### DISCUSSION TITLE
Failed reading sectors: BOOT_MAX_IMG_SECTORS=128 - too small?

### DISCUSSION THREAD

**Discussion by Sathishkumar-rk** [mcu-tools/mcuboot#1401] [Q&A]
Hi Team,

I was trying to use the default settings in MCUBoot configuration for **mimxrt1160_evk_cm7** Board. I can able to build and flash the binary without any issue but when I do run the program, below error message is observed **Failed reading sectors: BOOT_MAX_IMG_SECTORS=128 - too small?**

Is this value should be changed to any other value?

Regards,
Sathishkumar K

**Comment by d3zd3z** [comment]
If you look in `boot/zephyr/boards` you'll see some config fragment files some some of the other mimxrt devices. You can use one of these as a template, but may need to figure out the right configuration for you're particular partition table.

One question, are you booting by loading code from external flash into RAM, or are you using mcuboot for the internal flash?

**Reply by Sathishkumar-rk** [reply]
I was trying to use internal flash memory only and I can able to run the code if set the IMAGE SECTOR VALUE AS 1024. Is this suggested value to use it in project?

**Reply by d3zd3z** [reply]
It's just going to depend on the number of sectors used in the partitions defined. What RTOS are you building with? In Zephyr, it looks like the flash configuration is for an external flash.

BTW, unless you have a strong reason not to, I would highly recommend eliminating the scratch partition, making slot 0 1 sector larger, and enabling "swap move". It's better in pretty much every way over the traditional swap algorithm.

**Reply by Sathishkumar-rk** [reply]
Yes, now started using external flash with zephyr.

Thanks for the update, how do we enable the **swap move** option? Can you please point out where to change that?
